File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/ShipmentShippedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.time.OffsetDateTime;
import java.util.UUID;

public record ShipmentShippedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        OffsetDateTime occurredAt
) {
}

File: shipment-service/src/main/kotlin/com/paulopacifico/shipmentservice/messaging/api/ShipmentShippedEvent.kt
package com.paulopacifico.shipmentservice.messaging.api

import java.time.OffsetDateTime
import java.util.UUID

data class ShipmentShippedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val occurredAt: OffsetDateTime,
)
